src/audio/tdfb/tdfb.c

```diff
@@ -117,14 +162,39 @@ static void tdfb_free_delaylines(struct tdfb_comp_data *cd)
 		fir[i].delay = NULL;
 }
 
+static void *tdfb_filter_seek(struct sof_tdfb_config *config, int num_filters)
+{
+	struct sof_fir_coef_data *coef_data;
+	int i;
+	uint8_t *coefp = (uint8_t *)ASSUME_ALIGNED(&config->data[0], 2); /* 2 for 16 bits data */
+
+	/* Note: FIR coefficients are int16_t. An uint_8 type pointer coefp
+	 * is used for jumping in the flexible array member structs coef_data.
+	 */
+	for (i = 0; i < num_filters; i++) {
+		coef_data = (struct sof_fir_coef_data *)coefp;
+		coefp += sizeof(*coef_data) + sizeof(int16_t) * coef_data->length;
```

Comment:
this then becomes
```c
(void *)coefp += sizeof(*coef_data) + sizeof(int16_t) * coef_data->length;
```
since this data contains a container array with embedded data arrays..

Reply:
and here then `coefp = coef_data->coef + coef_data->length;`

Reply:
> and here then `coefp = coef_data->coef + coef_data->length;`

doesn't this miss out the struct sof_fir_coef_data container size addition ?

Reply:
I think @lyakh 's proposal is correct. Coef is the last flexible array member. Please correct me if this is not safe to do.